src/sdl2/joystick.rs

```diff
@@ -90,6 +90,43 @@ impl JoystickSubsystem {
 
 }
 
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+#[repr(i32)]
+pub enum PowerLevel {
+    Unknown = sys::SDL_JoystickPowerLevel::SDL_JOYSTICK_POWER_UNKNOWN as i32,
```

Comment:
`Unknown` is actually returned plenty of the time on several supported platforms!

As I said in the original post:
> At this stage, due to limitations in SDL2 itself, this method only returns a useful value for XInput devices on Windows, but hopefully SDL2 will improve that in the future! On unsupported platforms, this will currently consistently return `Ok` wrapping `sdl2::joystick::PowerLevel::Unknown` with no error.

Reply:
That I agree with, but you have this line in the Rust code: `if result != sys::SDL_JoystickPowerLevel::SDL_JOYSTICK_POWER_UNKNOWN`, meaning that technically, the method cannot ever return `Ok(Unknown)`. We basically have two choices here:

* Remove `Unknown` from the SDL2 enum since it can't ever be returned from power_level anyway, and it isn't used anywhere else.
* Mention that the `power_level` method cannot ever return `Ok(Unknown)` and so this pattern is unreachable.

I'd tend to go towards solution 1), and Have to_ll() return an `Option<PowerLevel>` (`None` would be the equivalent of `SDL_JoystickPowerLevel::SDL_JOYSTICK_POWER_UNKNOWN`)

Reply:
I think we’re talking past each other here.

The method _will_ return `Ok(Unknown)` if on a platform where there is no power data available; in the “else” block it checks if there is an error, and if there is no error, it will return `Ok(Unknown)`.

I like your proposal of using `Option<PowerLevel>` but I put it together this way as it matches the pattern for dealing with potential error states as established by [the axis method below](https://github.com/Rust-SDL2/rust-sdl2/pull/759/files#diff-7a2ec6e70f79ee03e075cbc761473e63R216).

If moving away from that pattern is acceptable, I’d be happy to update it! Given the error state this intends to handle _shouldn’t_ be accessible if used from safe code, this should be safe to do.

Reply:
You're totally right, I guess I wasn't focused enough in this code review and misread what your PR did exactly.

There is no need to change something that works perfectly fine already, so we're going to keep everything else as it is :) Thank you for the PR!